boa/contracts/vyper/vyper_contract.py

```diff
@@ -924,6 +931,128 @@ def eval(
+        #    "chainId": "0x01",
+        #    "codeformat": "solidity-standard-json-input",
+        #    "sourceCode": source_code,
+        #    "constructorArguements": self._ctor.bytecode.decode("utf-8") if self._ctor is not None else "",
+        #    "contractaddress": str(self.address),
+        #    "contractname": f"{self.filename}:{self.contract_name}",
+        #    "compilerversion": self.compiler_data.settings.compiler_version,
+        # }
+        body = {}
+
+        response = requests.post(
+            api_endpoint,
+            headers={"Content-Type":"multipart/form-data"},
+            json=body
+        )
+
+        if response.status_code == 200:
+            print(f"Successfully verified contract: {self.contract_name} at addresss: {self.address}")
+            return True
+        else:
+            print(f"Failed to verify contract: {self.contract_name} at addresss: {self.address}")
+            return False
+
+
+    def _verify_blockscout(self) -> bool:
+        # get API endpoint for blockscout explorer
+        api_url = os.getenv("BLOCKSCOUT_API_URL", "https://eth.blockscout.com")
+        api_endpoint = f"{api_url}/api/v2/smart-contracts/{str(self.address).lower()}/verification/via/vyper-standard-input"
+        # print(f"API endpoint: {api_endpoint}")
+
+        # constructing contract source code in JSON format
+        filename = f"{self.contract_name}.vy" if self.filename is None or self.filename == "<unknown>" else self.filename
+        standard_input_str = json.dumps({
+            "language": "Vyper",
+            "sources": {
+                filename: {
+                    "content": self.compiler_data.source_code
+                }
+            },
+        }, indent=4)
+        files = {
+            "files[0]": (filename, standard_input_str.encode("utf-8"), "application/json")
+        }
+        # print(f"Standard json input: {files}")
+
+        # constructing the request body for verification
+        settings = copy.copy(self.compiler_data.settings)
+        data = {
+            "compiler_version": settings.compiler_version if settings.compiler_version is not None else "v0.4.0+commit.e9db8d9f",
+            "license_type": "none",
+            "evm_version": settings.evm_version if settings.evm_version is not None else "cancun",
+        }
+        # print(f"Data: {data}")
+        
+        try:
+            response = requests.post(
+                api_endpoint,
+                data=data,
+                files=files,
+            )
+            # print(f"The whole response object: {response.json()}")
+
+            if response.status_code == 200:
+                print(f"Successfully verified contract: {self.contract_name} at addresss: {self.address}")
+                return True
+            else:
+                print(f"Failed to verify contract: {self.contract_name} at addresss: {self.address}")
+                return False
+        except Exception as error:
```

Comment:
please let exceptions bubble up :)

Reply:
Why not catch in the function itself, since it will affect others function calls

Reply:
because the caller deserves to know that the verification failed

Reply:
But I already return false in this case, also when the status code is not success as well

Reply:
applied as you suggested :)